src/test/regress/sql/rpt.sql

```diff
@@ -562,6 +562,50 @@ explain (costs off, verbose) insert into t2 (a, b) select i, random() from t;
 explain (costs off, verbose) select * from t1 where a in (select f(i) from t where i=a and f(i) > 0);
 -- ensure we do not break broadcast motion
 explain (costs off, verbose) select * from t1 where 1 <= ALL (select i from t group by i having random() > 0);
+reset enable_bitmapscan;
+reset enable_seqscan;
+set gp_cte_sharing = on;
+-- ensure we make motion when volatile function in target list of CTE
+explain (costs off, verbose) with cte as (
+    select a * random() as a from generate_series(1, 5) a
+)
+select * from cte join (select * from t1 join cte using(a))b using(a);
+set gp_cte_sharing = off;
+explain (costs off, verbose) with cte as (select a, a * random() from generate_series(1, 5) a)
+select * from cte join t1 using(a);
+reset gp_cte_sharing;
+-- ensure we make motion when volatile function in target list of union
+explain (costs off, verbose) select * from (
+    select random() as a from generate_series(1,5)
+    union
+    select random() as a from generate_series(1,5)
+)
+a join t1 on a.a = t1.a;
+-- ensure we make motion when volatile function in target list of subplan of multiset function
+explain (costs off, verbose) select * from (
+    SELECT count(*) as a FROM anytable_out( TABLE( SELECT random()::int from generate_series(1,5)a))
+) a join t1 using(a);
+
+-- start_ignore
+drop table if exists d;
+-- end_ignore
+create table d (b int, a int default 1) distributed by (b);
+insert into d select * from generate_series(0, 20) j;
+-- make sure that the tuples are on more than one segment.
+SELECT count(distinct(gp_segment_id)) > 1 from d;
```

Comment:
Why do we doubt in data distribution policy?  If we do, it's better to write the insertion, about whose distribution we can be confident.

Reply:
in the case of a demo cluster (3 segments), the distribution will be suitable and the data will be on all three segments. but in other cases it may be different. if we want to run tests on only 3 segments, then this check can be omitted.

Reply:
If you read several .out results,you will notice, that there is maximum 3 segments at each plan. I suppose all the tests are written in context of 3-segment cluster.

Reply:
So, remove this check.
Additionally extend the tests description and write that we want to apply volatile functions only on single segment, and write why

Reply:
About description, writing ` ensure we make motion when volatile function in target list of CTE` is uninformative. It's better to write that the result of the plan with volatile target list is taken from single segment (motion may be not created in some cases, and motion can be different itself).  Something like that.

About changed one, what does the first sentence means?  And maybe it would be better to put this description ad line 589, above the `drop table if exists d;`, where the test actually starts.
```
-- In the case of a volatile function in the target list in a query with scan over
-- a node with a General or SegmentGeneral locus. The function call should be made
-- on single segment, since it is assumed that nodes with such locus will give the
-- same result on all segments. So result should be count = 1
```

Reply:
About PR description:
1.  First sentence - grammar is wrong, can't understand the sentence.
General and segmentGeneral locus imply that if the corresponding slice
**is executed** in many different segments **should provide** (who?) the same result
data set.
2. **in the case** that if they are in a subquery **in this case**. rewrite the whole sentence too

Reply:
"plans" - first word in a sentence